src/core/include/openvino/core/any.hpp

```diff
@@ -210,14 +218,36 @@ struct Read<
     std::map<K, T, C, A>,
     typename std::enable_if<std::is_default_constructible<K>::value && std::is_default_constructible<T>::value>::type> {
     void operator()(std::istream& is, std::map<K, T, C, A>& map) const {
-        while (is.good()) {
-            std::string str;
-            is >> str;
-            auto k = from_string<K>(str);
-            is >> str;
-            auto v = from_string<T>(str);
-            map.emplace(std::move(k), std::move(v));
+        char c;
+
+        is >> c;
+        OPENVINO_ASSERT(c == '{', "Failed to parse std::map<K, T>. Starting symbols is not '{', it's ", c);
```

Comment:
Can we reuse `void Read<AnyMap>::operator()(std::istream& is, AnyMap& map) const {` here?

Reply:
> Can we reuse `void Read<AnyMap>::operator()(std::istream& is, AnyMap& map) const {` here?

It is implementation of Read itself, is not it?

Reply:
> Can we reuse

Potentially yes. Suggest to do it in a separate PR if no other major comments.